src/main/java/it/gov/pagopa/pu/fileshare/exception/FileshareExceptionHandler.java

```diff
@@ -32,6 +33,11 @@ public ResponseEntity<FileshareErrorDTO> handleFileStorageError(RuntimeException
     return handleFileshareErrorException(ex, request, HttpStatus.INTERNAL_SERVER_ERROR, CodeEnum.FILE_UPLOAD_ERROR);
   }
 
+  @ExceptionHandler(FileDecryptionException.class)
+  public ResponseEntity<String> handleFileDecryptionException(FileDecryptionException ex) {
+    return ResponseEntity.status(HttpStatus.NOT_FOUND).body(ex.getMessage());
+  }
```

Comment:
why an exception called FileDecryptionException should return a 404?
Probably the name of the exception is wrong or the point where you are throwing it...
What about FileNotFoundException?